### PR TITLE
[Silabs] Add Pump app

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Add Silicon Labs implementation of the pump-application

### CODE REVIEW

File: examples/pump-app/silabs/src/ZclCallbacks.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+using namespace chip::app::Clusters;
+
+/***** Function declarations *****/
+void OnOnOffPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value);
+void OnLevelControlPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value);
+
+/***** Helper functions *****/
+void OnOnOffPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    BitMask<chip::app::Clusters::PumpConfigurationAndControl::PumpStatusBitmap> pumpStatus;
```
Comment (bzbarsky-apple) [inline examples/pump-app/silabs/src/ZclCallbacks.cpp:38]: Why all the prefixes everywhere if you have the `using namespace` above?  Do one or the other; there's no point doing both.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/pump-app/silabs/src/AppTask.cpp [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    }
+
+    PumpMgr().SetCallbacks(ActionInitiated, ActionCompleted);
+
+    sPumpLED.Init(PUMP_LED);
+    sPumpLED.Set(!PumpMgr().IsStopped());
+
+// Update the LCD with the Stored value. Show QR Code if not provisioned
+#ifdef DISPLAY_ENABLED
+    // GetLCD().WriteDemoUI(LightMgr().IsLightOn());
```
Comment (jmartinez-silabs) [inline examples/pump-app/silabs/src/AppTask.cpp:105]: can be remove

File: examples/pump-app/silabs/src/AppTask.cpp [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        sAppTask.mSyncClusterToButtonAction = false;
+    }
+}
+
+void AppTask::UpdateClusterState(intptr_t context)
+{
+    // Set On/Off state
+    EmberStatus status;
+    bool onOffState = !PumpMgr().IsStopped();
+    status          = chip::app::Clusters::OnOff::Attributes::OnOff::Set(1, onOffState);
```
Comment (jmartinez-silabs) [inline examples/pump-app/silabs/src/AppTask.cpp:248]: you could use the define you created here for the endpoint argument
`#define PCC_CLUSTER_ENDPOINT 1`

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/pump-app/silabs/src/AppTask.cpp (hunk 2 above)
Comment (jepenven-silabs) [reply examples/pump-app/silabs/src/AppTask.cpp:105]: fixed

**Pull request merged**